engelsystem/src/test/kotlin/info/metadude/android/eventfahrplan/engelsystem/extensions/CallExtensionsTest.kt

```diff
@@ -0,0 +1,176 @@
+package info.metadude.android.eventfahrplan.engelsystem.extensions
+
+import com.google.common.truth.Truth.assertThat
+import com.squareup.moshi.Moshi
+import info.metadude.android.eventfahrplan.engelsystem.models.ShiftsResult
+import info.metadude.kotlin.library.engelsystem.EngelsystemService
+import info.metadude.kotlin.library.engelsystem.models.Shift
+import kotlinx.coroutines.runBlocking
+import okhttp3.mockwebserver.MockResponse
+import okhttp3.mockwebserver.MockWebServer
+import org.junit.After
+import org.junit.Before
+import org.junit.Test
+import retrofit2.Call
+import retrofit2.Callback
+import retrofit2.Response
+import retrofit2.Retrofit
+import retrofit2.converter.moshi.MoshiConverterFactory
+import java.io.EOFException
+import java.net.HttpURLConnection
+
+class CallExtensionsTest {
+
+    companion object {
+        private const val VALID_SHIFTS_JSON = "[{\"Comment\":\"This is a very secret comment.\",\"Name\":\"Kirmes\",\"RID\":12,\"SID\":579,\"TID\":10,\"UID\":32,\"created_at_timestamp\":1565979486,\"created_by_user_id\":32,\"description\":\"Kirmes are fun.\",\"edited_at_timestamp\":1567367095,\"edited_by_user_id\":32,\"freeloaded\":0,\"id\":37,\"name\":\"Collect stickers\",\"shifttype_id\":6,\"title\":\"Tag 1: Decorate fridge\"}]"
+        private val ONE_ITEM_SHIFTS = listOf(Shift(
+                userComment = "This is a very secret comment.",
+                locationDescription = "Kirmes are fun.",
+                locationName = "Kirmes",
+                name = "Collect stickers",
+                sID = 579,
+                talkTitle = "Tag 1: Decorate fridge",
+                typeId = 6
+        ))
+    }
+
+    private val mockWebServer = MockWebServer()
+
+    @Before
+    fun setUp() {
+        mockWebServer.start()
+    }
+
+    @After
+    fun tearDown() {
+        mockWebServer.shutdown()
+    }
+
+    @Test
+    fun `awaitShiftsResult returns success when call responds with shifts list`() {
+        withSuccessResponse(Response.success(ONE_ITEM_SHIFTS)) { call ->
+            val shiftsResult = call.awaitShiftsResult()
+            assertThat(shiftsResult).isEqualTo(ShiftsResult.Success(ONE_ITEM_SHIFTS))
+        }
+    }
+
+    @Test
+    fun `awaitShiftsResult returns MissingResponseSuccessBody exception when call responds with null body`() {
+        withSuccessResponse(Response.success(null)) { call ->
+            val shiftsResult = call.awaitShiftsResult()
+            assertThat(shiftsResult).isEqualTo(ShiftsResult.Exception.MissingResponseSuccessBody)
+        }
+    }
+
+    @Test
+    fun `awaitShiftsResult returns success when call responds with HTTP 200`() {
+        withResponse(HttpURLConnection.HTTP_OK) { call ->
+            val shiftsResult = call.awaitShiftsResult()
+            assertThat(shiftsResult).isEqualTo(ShiftsResult.Success(ONE_ITEM_SHIFTS))
```

Comment:
If you assert against `ONE_ITEM_SHIFTS` I'd pass `VALID_SHIFTS_JSON` to `withResponse`. I try to only use default arguments for parameters that are not relevant to the test. 
By only reading the test and not the implementation of `withResponse()` the reader has no idea why the result should be `ONE_ITEM_SHIFTS`.

Side note: Maybe the names of the constants should be changed so it's clear it's the same data in two different representations.

Reply:
:heavy_check_mark: I thought about this, too. Changed in 0834d19ca714e71a4ebb347a4ccbe518625b3bfe.